File: <100/059/059.py
def openfile(filename: str) -> list[int]:
    with open(filename, 'r') as f:
        s: str = f.readline()
    num_chars: list[str] = s.split(',')
    numbers = map(int, num_chars)

    return list(numbers)


def xor_s(numbers: list[int], keys: list[int]) -> str:
    nl = len(numbers)
    kl = len(keys)
    result = ['' for _ in range(nl)]
    for i in range(nl):
        result[i] = chr(numbers[i] ^ keys[i%kl] + ord('a'))
    
    return "".join(result)


def sum_message(message: str) -> int:
    return sum(ord(c) for c in message)


def solve(filename: str) -> int:
    numbers = openfile(filename)
    for k1 in range(26):
        for k2 in range(26):
            for k3 in range(26):
                message = xor_s(numbers, [k1, k2, k3])
                if "Euler" in message:
                    print(message)
                    return sum_message(message)
    return -1


def main() -> None:
    filename: str = "059.txt"
    answer = solve(filename)
    return answer


if __name__ == '__main__':
    answer = main()
    print(answer)

File: <100/059/059.jl
function read_file(filename:: String) :: Vector{Int}
  ls = filename |> readline |> (x -> split(x, ","))
  vc = map(String, ls)
  vn = map(x -> parse(Int, x), vc)
  return vn 
end


function xor_s(numbers :: Vector{Int}, keys :: Vector{Int}) :: String
  keys = map(x -> x + Int('a'), keys)
  len = numbers |> length
  nk  = keys |> length
  vc :: Vector{Char} = ['a' for _ ∈ 1:len]
  for i ∈ 1:len
    xored = ⊻(numbers[i], keys[begin + i%nk])
    vc[i] = Char(xored)
  end

  return String(vc)
end


function sum_message(message :: String) :: Int
  chars = collect(message)
  ints = map(Int, chars)
  return sum(ints)
end


function solve(filename :: String) :: Int
  numbers = filename |> read_file
  for keys ∈ 0:(26^3-1)
    k1 = keys % 26
    k2 = (keys ÷ 26) % 26
    k3 = (keys ÷ 26^2) % 26
    message :: String = xor_s(numbers, [k1, k2, k3])
    if occursin("Euler", message)
      message |> println
      return sum_message(message)
    end
  end

  return -1
end


function main() :: Nothing
  filename :: String = "059.txt"
  filename |> solve |> println
end


@time main()
